lua/astrocommunity/pack/typescript/biome.lua

```diff
@@ -0,0 +1,32 @@
+local has_biome = function()
+  local bufnr = vim.api.nvim_get_current_buf()
+  local rooter = require "astrocore.rooter"
+
+  local biomejson_rooter = rooter.resolve {
+    "biome.json",
+  }
+
+  return next(biomejson_rooter(bufnr))
+end
+
+return {
+  { import = "astrocommunity.pack.typescript.core" },
+  {
+    "williamboman/mason-lspconfig.nvim",
+    opts = function(_, opts)
+      opts.ensure_installed = require("astrocore").list_insert_unique(opts.ensure_installed, { "biome" })
+    end,
+  },
+  {
+    "jay-babu/mason-null-ls.nvim",
+    opts = function(_, opts)
+      if not opts.handlers then opts.handlers = {} end
+
+      if not has_biome() then return end
+
+      -- TODO: prettier should be off for all filetypes that biome supports,
+      -- currently this is { "javascript", "javascriptreact", "typescript", "typescriptreact", "json" }
+      opts.handlers.prettierd = function() end
```

Comment:
I don't know much about configuring null-ls, how would you do this?

The reason for wanting to do this is that prettierd will attach to `json` files even though biome is running and should be the primary formatter, but for markdown it should still be prettier.